src/celeritas/optical/ImportedOpticalMaterials.hh

```diff
@@ -0,0 +1,65 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/ImportedOpticalMaterials.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "celeritas/io/ImportOpticalMaterial.hh"
+#include "celeritas/optical/Types.hh"
+
+namespace celeritas
+{
+struct ImportData;
+//---------------------------------------------------------------------------//
+/*!
+ * A registry for imported optical material data.
```

Comment:
In that case I think I'd like to change the optical map in `ImportData` to be a vector of `ImportOpticalMaterial`s and change the map to be a map between `GeoMatId`s and `OpticalMaterialId`s. We need to just iterate over a list of optical materials in several places, and having a simple vector to index would be convenient.

Reply:
@hhollenb I fully support that, it's definitely more inline with how we treat the other properties. Perhaps instead of a map between geo and optical matIDs, what about having an `unsigned int optical_material_id` field in `ImportGeoMaterial`? (You could define a `static inline constexpr unsigned int unspecified = -1` like we have with `ImportVolume`.)